draft-irtf-cfrg-opaque.md

```diff
@@ -393,152 +397,342 @@ This specification defines one particular AKE based on 3DH; see {{instantiations
 We let `Npk` and `Nsk` denote the size of public and private keys, respectively,
 used in the AKE.
 
-# Offline Registration {#offline-phase}
+- Deterministic Key Generation Functions:
+  - RecoverPublicKey(private_key): Recover the public key related to the input private key.
+  - DeriveKeyPair(seed): Derive a private and public key pair deterministically from the seed.
 
-Registration is executed between a client C and a
-server S. It is assumed S can identify C and the client can
-authenticate S during this registration phase. This is the only part
-in OPAQUE that requires an authenticated and confidential channel, either physical, out-of-band,
-PKI-based, etc. This section describes the registration flow, message encoding,
-and helper functions. Moreover, C has a key pair (client_private_key, client_public_key) for an AKE protocol
-which is suitable for use with OPAQUE; See {{online-phase}}. The private-public keys (client_private_key, client_public_key) may be randomly generated (using a cryptographically secure pseudorandom number generator) for the account or provided by the calling client.
-Clients MUST NOT use the same key pair (client_private_key, client_public_key) for two different accounts.
+Generally, nonces are of length `Nn` set to 32 bytes.
```

Comment:
Can we remove the word "generally"? Or are there exceptions to this rule? Also, is there ever a case where `Nn` is not equal to 32, and if so, can we just remove the use of this extra variable and hard-code 32 in place of `Nn`?

Reply:
+1 for dropping generally

On the other hand, I'm less comfortable with hardcoding values. But if you both think we should revert to do it, I can live with that :)

Reply:
I just saw @chris-wood  suggestion !